atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -168,6 +192,31 @@ def _pull_and_tag_image(self, image, build_json, nonce):
         self.log.error("giving up trying to pull image")
         raise RuntimeError("too many attempts to pull and tag image")
 
+    def _get_manifest_list(self, image):
```

Comment:
We can make this a bit more efficient by remembering the manifest list for each image:
```python
def _get_manifest_list(self, image, cache={}):
  if image not in cache:
    ...
    cache[image] = manifest_list
  return cache[image]
```

Reply:
In future we could consider decorating with @functools.lru_cache instead of handling caching explicitly, since atomic-reactor does not need to run on Python < 2.7.